server/src/main/java/keywhiz/service/providers/ClientAuthFactory.java

```diff
@@ -293,7 +293,7 @@ private boolean containsUri(List<String> uriList, URI targetUri) {
    * @return whether to create a client that can't be found, if the client doesn't exist
    */
   protected boolean createMissingClient() {
-    return true;
+    return clientAuthConfig.createMissingClients();
```

Comment:
Is it necessary to have both `createMissingClients()` and `createMissingClient()`? Since createMissingClient() ends up calling `createMissingClients()`anyway.

Reply:
Yes; the `createMissingClient()` method gets overridden by the `AutomationClientAuthFactory`, because that factory never creates clients ([here](https://github.com/square/keywhiz/blob/2268480482fdfbf7389aded803605b01ebc3d583/server/src/main/java/keywhiz/service/providers/AutomationClientAuthFactory.java#L64-L68))

Reply:
Thanks for clarifying!